team_project_access.go

```diff
@@ -57,14 +58,106 @@ type TeamProjectAccessList struct {
 
 // TeamProjectAccess represents a project access for a team
 type TeamProjectAccess struct {
-	ID     string                `jsonapi:"primary,team-projects"`
-	Access TeamProjectAccessType `jsonapi:"attr,access"`
+	ID              string                      `jsonapi:"primary,team-projects"`
+	Access          TeamProjectAccessType       `jsonapi:"attr,access"`
+	ProjectAccess   *ProjectAccessPermissions   `jsonapi:"attr,project-access"`
+	WorkspaceAccess *WorkspaceAccessPermissions `jsonapi:"attr,workspace-access"`
 
 	// Relations
 	Team    *Team    `jsonapi:"relation,team"`
 	Project *Project `jsonapi:"relation,project"`
 }
 
+// ProjectAccessPermissions represents the team's specific permissions related to the project itself
+type ProjectAccessPermissions struct {
+	ProjectSettingsPermission ProjectSettingsPermissionType `jsonapi:"attr,settings"`
+	ProjectTeamsPermission    ProjectTeamsPermissionType    `jsonapi:"attr,teams"`
+}
+
+// WorkspaceAccessPermissions represents the team's specific permissions related to workspaces in the project
+type WorkspaceAccessPermissions struct {
+	WorkspaceRunsPermission          WorkspaceRunsPermissionType          `jsonapi:"attr,runs"`
+	WorkspaceSentinelMocksPermission WorkspaceSentinelMocksPermissionType `jsonapi:"attr,sentinel-mocks"`
+	WorkspaceStateVersionsPermission WorkspaceStateVersionsPermissionType `jsonapi:"attr,state-versions"`
+	WorkspaceVariablesPermission     WorkspaceVariablesPermissionType     `jsonapi:"attr,variables"`
+	WorkspaceCreatePermission        bool                                 `jsonapi:"attr,create"`
+	WorkspaceLockingPermission       bool                                 `jsonapi:"attr,locking"`
+	WorkspaceMovePermission          bool                                 `jsonapi:"attr,move"`
+	WorkspaceDeletePermission        bool                                 `jsonapi:"attr,delete"`
+	WorkspaceRunTasksPermission      bool                                 `jsonapi:"attr,run-tasks"`
+}
+
+// ProjectSettingsPermissionType represents the permissiontype to a project's settings
+type ProjectSettingsPermissionType string
+
+const (
+	ProjectSettingsPermissionRead   ProjectSettingsPermissionType = "read"
+	ProjectSettingsPermissionUpdate ProjectSettingsPermissionType = "update"
+	ProjectSettingsPermissionDelete ProjectSettingsPermissionType = "delete"
+)
```

Comment:
Ditto for the rest of the types defined here -- should be organized under their respective resources even if only used by the `TeamProjectAccess` struct fields.

Reply:
This would break with what is already done on the TeamWorkspace level. While this file is called team access which is a little confusing, here there is the similar pattern of the workspace level permissions inside the access class itself: https://github.com/hashicorp/terraform-provider-tfe/blob/main/tfe/resource_tfe_team_access.go#L62-L123
Moving this would also break with the API design since you set permission on the TeamProject and not the Project itself.